src/pudl/transform/eia_bulk_elec.py

```diff
@@ -0,0 +1,298 @@
+"""Clean and normalize EIA bulk electricity data."""
+from functools import reduce
+from io import StringIO
+
+import pandas as pd
+
+# modified from:
+# meta.groupby(["sector", "sector_code"])['description'].first().str.strip('; ').str.split(';', expand=True)
+EIA_SECTOR_AGGREGATE_ASSOCIATION = pd.read_csv(
+    StringIO(
+        """
+sector,is_cogen,aggregate,sector_code_eia,aggregate_description_eia
+commercial,True,all sectors,99,All sectors
+commercial,False,all sectors,99,All sectors
+industrial,True,all sectors,99,All sectors
+industrial,False,all sectors,99,All sectors
+electric utility,True,all sectors,99,All sectors
+electric utility,False,all sectors,99,All sectors
+independent power producers,True,all sectors,99,All sectors
+independent power producers,False,all sectors,99,All sectors
+industrial,True,all industrial (total),97,Power plants in the industrial sector
+industrial,False,all industrial (total),97,Power plants in the industrial sector
+commercial,True,all commercial (total),96,Power plants in the commercial sector
+commercial,False,all commercial (total),96,Power plants in the commercial sector
+electric utility,True,electric utility,1,Power plants owned by regulated electric utilties
+electric utility,False,electric utility,1,Power plants owned by regulated electric utilties
+independent power producers,True,independent power producers (total),94,Power plants owned by unregulated power companies (also called merchant generators)
+independent power producers,False,independent power producers (total),94,Power plants owned by unregulated power companies (also called merchant generators)
+electric utility,True,electric power (total),98,Power plants owned by companies whose primary purpose is to produce power
+electric utility,False,electric power (total),98,Power plants owned by companies whose primary purpose is to produce power
+independent power producers,True,electric power (total),98,Power plants owned by companies whose primary purpose is to produce power
+independent power producers,False,electric power (total),98,Power plants owned by companies whose primary purpose is to produce power
+"""
+    ),
+)
+"""Association table describing the many-to-many relationships between plant sectors and various levels of EIA's aggregates."""
+
+
+# modified from:
+# meta.groupby(["fuel", "sector_code"])['description'].first().str.strip('; ').str.split(';', expand=True)
+# PEL is defined as "Summation of all petroleum liquids (distallte fuel oil, jet fuel, residual fuel oil, kerosense waste oil and other petroleum liquids)"
+# Missing from this list are all the "other" categories of gases: OG, BFG, SGP, SC, PG
+# Those gases combine for about the same total MMBTU as DFO, about 0.2% of all reported fuel receipts
+EIA_FUEL_AGGREGATE_ASSOCIATION = pd.read_csv(
```

Comment:
The EIA bulk data contains aggregates of the more granular EIA data present in PUDL. EIA identifies these aggregates using codes unique to each aggregate (eg "all types of coal" = "COW"). PUDL has never seen some of these codes before because they only apply to aggregates found in the bulk data. But sometimes the aggregate is over a single fuel, so the aggregate code happens to be the same as the existing `energy_source_code` present in PUDL (eg the aggregate over all bitumenous coal is identified with "BIT", which is also the `energy_source_code` for bitumenous coal). The bulk data contains multiple levels of aggregation, so the same fuel may be present in multiple aggregates (eg the BIT fuel is in both the BIT aggregate and the COW aggregate).

The table `EIA_FUEL_AGGREGATE_ASSOCIATION` defines all these aggregates in terms of the `energy_source_code` values contained in that aggregate. It is a many-to-many mapping, AKA an "association table". The boiler-generator association table is another example of such a mapping. This mapping enables us to join aggregate codes onto the `fuel_receipts_costs_eia923` table so we can identify which aggregates apply to each entry.

I don't think we should combine this table with the `energy_sources_eia` table because this table describes many-to-many relationships and thus must contain duplicates of some keys. That would make it harder to look up simple information about fuel types, which is the purpose of the `energy_sources_eia` table.